src/main/java/com/artipie/asto/memory/InMemoryStorage.java

```diff
@@ -89,19 +90,17 @@ public InMemoryStorage() {
 
     @Override
     public CompletableFuture<Void> save(final Key key, final Content content) {
-        return CompletableFuture.runAsync(
-            () -> {
-                synchronized (this.data) {
-                    this.data.put(
-                        key.string(),
-                        new Remaining(
-                            new Concatenation(content).single().blockingGet(),
-                            true
-                        ).bytes()
-                    );
+        return new Concatenation(content).single().to(SingleInterop.get())
```

Comment:
@olegmoz can we add a test that will verify that the bug is fixed?

Reply:
@Vatavuk though about it, but not sure how to do it. The problem is we need to check that code is not getting stuck in synchronized section. But what if it still does? The unit test will never finish. Not sure that having such test is a nice idea

Reply:
@olegmoz maybe you can use `@Timeout` junit annotation? It will fail the test if it can't pass within specified amount of time, e.g. this test will fail:
```java
@Test
@Timeout(value = 1, unit = TimeUnit.SECONDS)
void test() throws Exception {
    Thread.sleep(2000);
}
```
with error:
> java.util.concurrent.TimeoutException: test() timed out after 1 second

Reply:
@olegmoz maybe we can run the test code in a separate thread and kill the thread if more than X seconds passed and fail the test. If it is too complex and you don't see an additional value that we would get from such a test than we can leave as is.

Reply:
@g4s8 @Vatavuk wqs pretty sure timeout feature was still missing in junit, thanks! Added the test, please review